v2/internal/frontend/desktop/linux/frontend.go

```diff
@@ -9,6 +9,69 @@ package linux
 #include "gtk/gtk.h"
 #include "webkit2/webkit2.h"
 
+// CREDIT: https://github.com/rainycape/magick
+#include <errno.h>
+#include <signal.h>
+#include <stdio.h>
+#include <string.h>
+
+static void fix_signal(int signum)
+{
+    struct sigaction st;
+
+    if (sigaction(signum, NULL, &st) < 0) {
+        goto fix_signal_error;
+    }
+    st.sa_flags |= SA_ONSTACK;
```

Comment:
Should we skip fixup the signal handler if the `SA_ONSTACK` flag is already set?

Reply:
Yeah absolutely, would maybe just save some cycles for internal setting up the handler in the kernel when it's already fine.But that can be neglected 😄